drivers/serial/uart_npcx.c

```diff
@@ -102,42 +101,43 @@ static int uart_set_npcx_baud_rate(struct uart_reg *const inst, int baud_rate, i
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
 static int uart_npcx_tx_fifo_ready(const struct device *dev)
 {
-	struct uart_reg *const inst = HAL_INSTANCE(dev);
+	const struct uart_npcx_config *const config = dev->config;
```

Comment:
How about:
struct uart_reg *const reg_base =dev->config->uart;
So we can just access the UART register like:
GET_FIELD(reg_base->UFTSTS, NPCX_UFTSTS_TEMPTY_LVL)